### PR TITLE
DEV-45405 - Fix grafana tests

### PR DESCRIPTION
**Pull request by yasmin-tr**
fix the failing grafana unit tests so we can have green pipeline

tests fixed:
* api_notification_channel_test.go - fixed to expected strings returned from notifications
* error_handling_test.go - fixed expected response due to our change in errors
* api_plugins_test.go - updated the expectedListResp.json
* dashboard_test.go - TestDashboardVersionsAPIEndpoint
* templates_test.go - removed orgId from urls

also:
* changed implementation in alertmanager.go NewAlertmanager() to avoid null exception if there is no ParsedAppUrl provided in config (fixes a lot of tests)
* changed implementation in navtree.go to remove the connections section (datasources) in another place, to help tests pass and makes more sense.

### CODE REVIEW

**Comment by yasmin-tr** [comment]
run-test

**Comment by npololnskii** [comment]
run-test

**Comment by yasmin-tr** [comment]
run-test

**Comment by yasmin-tr** [comment]
run-test

**Comment by npololnskii** [comment]
run-test

**Comment by npololnskii** [comment]
run-test

**Comment by npololnskii** [comment]
run-test

**Comment by yasmin-tr** [comment]
run-test

**Comment by yasmin-tr** [comment]
run-test

**Review by copyhold — APPROVED** [review]
i hope you checked that datasource menu is really not visible...

File: pkg/services/navtree/navtreeimpl/navtree.go [hunk 1]
```diff
@@ -485,7 +487,7 @@ func (s *ServiceImpl) buildDataConnectionsNavLink(c *contextmodel.ReqContext) *n
 
 	baseUrl := s.cfg.AppSubURL + "/connections"
 
-	if false && hasAccess(datasources.ConfigurationPageAccess) { // LOGZ.IO GRAFANA CHANGE :: hide datasources menu, should be done by altering permissions
+	if hasAccess(datasources.ConfigurationPageAccess) {
```
Comment (Jonathan-Eng) [inline pkg/services/navtree/navtreeimpl/navtree.go:490]: this has no Grafana update msg?

**Review by Jonathan-Eng — APPROVED** [review]

**Comment by yasmin-tr** [comment]
run-test

File: pkg/services/navtree/navtreeimpl/navtree.go (hunk 1 above)
Comment (yasmin-tr) [reply pkg/services/navtree/navtreeimpl/navtree.go:490]: this is a revert to original code, that's why i removed the logz comment (i simply moved the logz change to the part that calls this function to make the tests pass 🙏 )

**Pull request merged**